### DISCUSSION TITLE
Translator can't pickup stream sound

### DISCUSSION THREAD

**Discussion by nicogb2007** [cyberofficial/Synthalingua#12] [Q&A] [answered]
The translator is not picking the sound for some reasons.
I've already put it so it listens on the headset sound.
![image](https://user-images.githubusercontent.com/107760410/229322780-48be4070-6abf-4f36-b2b6-9afe990fe7ce.png)
here is the command line i use to set my mic as mice for the translator
`python transcribe_audio.py --set_microphone 8`
Here is the complete list of mics via the command 
`python transcribe_audio.py --list_microphones`
![image](https://user-images.githubusercontent.com/107760410/229322822-6246be42-4182-493f-bd45-3cdf88112c5a.png)

Can someone help me with that issue ?

**Comment by cyberofficial** [comment, marked as answer]
Which windows are you on?

**Reply by nicogb2007** [reply]
i'm on windows 10

**Reply by cyberofficial** [reply]
If you have stereo mix in your recording menu, you could use that to capture desktop audio
<img width="310" alt="parsecd_hs8cgcQO6Z" src="https://user-images.githubusercontent.com/19499442/229323517-0f3b52c4-1b70-45e3-bfa0-699096276edb.png">

You could also install Voice Meter to rout audio as well.

**Reply by nicogb2007** [reply]
It should work if i configure it like that ?
![image](https://user-images.githubusercontent.com/107760410/229323550-9b2357ba-abb9-4f11-8dd6-3a749b22da64.png)

**Reply by nicogb2007** [reply]
I think it's working. It's picking up sound but it's not translating it
![image](https://user-images.githubusercontent.com/107760410/229323816-85e415da-0988-4354-969e-17857246ea60.png)

I just restarted it and it wont pickup audio

**Reply by cyberofficial** [reply]
<img width="722" alt="parsecd_IwtlayykPj" src="https://user-images.githubusercontent.com/19499442/229324362-3357fdbc-72c4-4ea4-afd7-11ae9539712c.png">

Make sure you set the device id of the mix with the `--set_microphone` flag ex: `--set_microphone 2`

(repo was also updated to fixed some bugs and add a better table for audio)

**Reply by nicogb2007** [reply]
Did exactly that but it does not detect stream audio
![image](https://user-images.githubusercontent.com/107760410/229324464-7f92e922-81ad-4d85-8f8c-447b1a326e4e.png)
![image](https://user-images.githubusercontent.com/107760410/229324485-9b585a28-4395-489b-880b-c5de77dd1678.png)

**Reply by cyberofficial** [reply]
Does the audio source show that there is audio? Like so?
<img width="596" alt="parsecd_YX7GTKyKjV" src="https://user-images.githubusercontent.com/19499442/229324530-17f0fb01-a5fc-4f33-9e89-8e5a1ae53462.png">

**Reply by nicogb2007** [reply]
Nope it does not :/
![image](https://user-images.githubusercontent.com/107760410/229324555-d4d6b4c7-8b79-4ac5-801a-0a71f2b1cc62.png)

**Reply by cyberofficial** [reply]
is it enabled?

**Reply by nicogb2007** [reply]
Yes
![image](https://user-images.githubusercontent.com/107760410/229324584-06568df9-09f9-44c8-9855-0c2271bf7e6a.png)

**Reply by cyberofficial** [reply]
You may need to get voice meter then and rout audio that way. You may have an audio setting conflicting with it.

They have a free one call voice meter banana. https://vb-audio.com/Voicemeeter/banana.htm

You can rout audio out to an extra audio cable and you can capture audio that way. It's what most people will choose to do anyway for recording sound.

If you need help setting up, I can help

**Reply by nicogb2007** [reply]
I would like some help as i dont know how to do the config

**Reply by cyberofficial** [reply]
Do you have it downloaded now?

**Reply by nicogb2007** [reply]
yes
![image](https://user-images.githubusercontent.com/107760410/229325684-6210094e-3829-47f7-b1e4-4e7541b42a26.png)

**Reply by cyberofficial** [reply]
Thats normal voice metere, close that out and open banana 
![image](https://user-images.githubusercontent.com/19499442/229325874-25cdd473-e1d1-4820-8dc0-b68a50d97582.png)
it will say banana on the top left

**Reply by nicogb2007** [reply]
found it 
![image](https://user-images.githubusercontent.com/107760410/229325933-2b8e06c8-8927-436f-a615-10f20877286d.png)

**Reply by cyberofficial** [reply]
Neat. Okay

Steps.
Uncheck all of these 
<img width="320" alt="opera_JZkdi9WOU0" src="https://user-images.githubusercontent.com/19499442/229326392-30ff1400-0124-4cf6-aaf4-993092b201a5.png">

set your headphones in A1
<img width="301" alt="opera_bhn3cl0oYS" src="https://user-images.githubusercontent.com/19499442/229326408-37883441-946b-4066-ac6a-30b584990515.png">

in sound, sound VoiceMeeter input as default in Play Back
<img width="188" alt="parsecd_PHH4n2ExF0" src="https://user-images.githubusercontent.com/19499442/229326425-c65a3468-d1a3-409b-800a-d8db4e0ee7ab.png">

In Recording tab, set VoiceMeeter Aux Output as default device
<img width="191" alt="opera_OBgt2otNcQ" src="https://user-images.githubusercontent.com/19499442/229326451-1a69b1a5-a278-48c7-ac62-34ab2168ec4c.png">
.png

Set B2 Under VAIO column
![image](https://user-images.githubusercontent.com/19499442/229326473-5112e163-1138-45dd-9e4c-d1ada2b1a5e7.png)

then look for the id of "VoiceMeeter Output (VB-Audio Vo"
<img width="801" alt="parsecd_8YPsGDYA4z" src="https://user-images.githubusercontent.com/19499442/229326761-505e258a-eb14-45b9-ad7b-08f8e8112269.png">



and then run the script with the right arguments. it should work

**Reply by cyberofficial** [reply]
There may be a few VoiceMeeter options, you may need to find the one that works for you

**Reply by nicogb2007** [reply]
It works thanks you !!!
But is there any settings i need to change on discord or anything ?

**Reply by nicogb2007** [reply]
and btw do you know why i can't run the 6Go ver 
![image](https://user-images.githubusercontent.com/107760410/229326877-75c23703-69f7-4739-af91-cd1dab0f6ad9.png)
My graphics card got 6Go of vram

**Reply by cyberofficial** [reply]
> It works thanks you !!! But is there any settings i need to change on discord or anything ?

Yea you need to change discord input and out in settings to use the new audio devices, need to tinker with it

Also i updated the script use the `--ramforce` flag to try and force it to use the 6gb flag you have set

**Reply by cyberofficial** [reply]
also make sure you use `6gb` as is, there's no support for other languages yes.

**Reply by nicogb2007** [reply]
it does not start with the `--rameforce 6gb`
that is the thing i get using `--ram 6gb` that might be why it does not work
![image](https://user-images.githubusercontent.com/107760410/229327642-b1120815-4ce6-443d-bdf4-ef2e8ca26c68.png)

**Reply by cyberofficial** [reply]
its just `--ramforce` like this `python .\transcribe_audio.py --set_microphone 1 --ram 6gb --translate --language ja --ramforce`

**Reply by nicogb2007** [reply]
It crashes on launch

**Reply by nicogb2007** [reply]
btw is the fine tuning completly done for every talent or do you plan to add an option for fine tunning for every talent ?

**Reply by cyberofficial** [reply]
you should be able to use the 6gb, i can use it on my 2060 just fine
![image](https://user-images.githubusercontent.com/19499442/229327852-4b8e2cf1-7150-4b98-aa0b-9ef252ec44a2.png)


> It crashes on launch

make sure you have the python environment activated `data_whisper\Scripts\activate` or `data_whisper\Scripts\activate.bat` in the same folder as the script in a console window

**Reply by cyberofficial** [reply]
> btw is the fine tuning completly done for every talent or do you plan to add an option for fine tunning for every talent ?

It's done for everyone

**Reply by nicogb2007** [reply]
It's already here by default but i dont get the activation message
call "C:\Users\*****\OneDrive\Bureau\Nouveau dossier (4)\Synthalingua-master\Synthalingua-master\data_whisper\Scripts\activate.bat"
![image](https://user-images.githubusercontent.com/107760410/229346834-164ef3f7-1485-4c01-b1f0-8c8965d1e667.png)

**Reply by cyberofficial** [reply]
Did you add the microphone flag to the file?

**Reply by nicogb2007** [reply]
This is the startup bat i use
```
@echo off 
cls 
call "C:\Users\*******\OneDrive\Bureau\Nouveau dossier (4)\Synthalingua-master\Synthalingua-master\data_whisper\Scripts\activate.bat" 
python "C:\Users\******\OneDrive\Bureau\Nouveau dossier (4)\Synthalingua-master\Synthalingua-master\transcribe_audio.py" --ram 6gb --non_english --translate --language ja --record_timeout 1 --discord_webhook "https://discord.com/api/webhooks/10918********85865/lz-5MyZED********tEptNH5HpHm9-68kJbuPas6OiiH14gux8BC3Dvx4Hza40" --energy_threshold 500 --set_microphone 1
```

**Reply by cyberofficial** [reply]
in the voice meeter app, does it show the sound mixer moving when playing audio?
![image](https://user-images.githubusercontent.com/19499442/229349351-5546bf60-66a1-4a27-a700-a0cf3a583fc1.png)

**Reply by nicogb2007** [reply]
Yup it shows the sound in voicemeter
The program works but for some reason it can't run the 6gb ver

**Reply by nicogb2007** [reply]
I just installed the last ver and now it works perfectly fine with the 6gb ver
